src/plots/cartesian/layout_attributes.js

```diff
@@ -137,34 +137,43 @@ module.exports = {
         ].join(' ')
     },
     tick0: {
-        valType: 'number',
+        valType: 'any',
         dflt: 0,
         role: 'style',
         description: [
             'Sets the placement of the first tick on this axis.',
             'Use with `dtick`.',
             'If the axis `type` is *log*, then you must take the log of your starting tick',
-            '(e.g. to set the starting tick to 100, set the `tick0` to 2).',
+            '(e.g. to set the starting tick to 100, set the `tick0` to 2)',
+            'except when `dtick`=*L<f>* (see `dtick` for more info).',
             'If the axis `type` is *date*, it should be a date string, like date data.',
             'If the axis `type` is *category*, it should be a number, using the scale where',
             'each category is assigned a serial number from zero in the order it appears.'
         ].join(' ')
     },
     dtick: {
         valType: 'any',
-        dflt: 1,
         role: 'style',
         description: [
-            'Sets the step in-between ticks on this axis',
-            'Use with `tick0`.',
+            'Sets the step in-between ticks on this axis. Use with `tick0`.',
+            'Must be a positive number, or special strings available to *log* and *date* axes.',
             'If the axis `type` is *log*, then ticks are set every 10^(n*dtick) where n',
             'is the tick number. For example,',
             'to set a tick mark at 1, 10, 100, 1000, ... set dtick to 1.',
             'To set tick marks at 1, 100, 10000, ... set dtick to 2.',
             'To set tick marks at 1, 5, 25, 125, 625, 3125, ... set dtick to log_10(5), or 0.69897000433.',
+            '*log* has several special values; *L<f>*, where `f` is a positive number,',
+            'gives ticks linearly spaced in value (but not position).',
+            'For example `tick0` = 0.1, `dtick` = *L0.5* will put ticks at 0.1, 0.6, 1.1, 1.6 etc.',
+            'To show powers of 10 plus small digits between, use *D1* (all digits) or *D2* (only 2 and 5).',
+            '`tick0` is ignored for *D1* and *D2*.',
```

Comment:
These have always existed behind the scenes but I think there were some bugs if you tried to use them yourself as opposed to getting them automatically. Now they work manually, and are tested at the level of supplyDefaults (which is where the bugs were before, I think) though at some point it might be worth adding tests of the functionality when used manually.